pkg/handler/repositories.go

```diff
@@ -313,7 +313,7 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error updating repository", err.Error())
 	}
 
-	if repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
+	if urlUpdated && repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
```

Comment:
it seems like `urlUpdated` already checks `repoParams.URL != nil && repoConfig.URL != *repoParams.URL`, so I think you can remove everything to the right of `urlUpdated`.

Reply:
You are correct good sir! 
Thank you.